src/test/java/com/google/sps/SentimentServletTest.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps;
```

Comment:
Comment for the whole test:

You want to test behavior of your servlet and not the helper methods. I recommend reading ToTTs `Test Behaviors, Not Methods` and `Test Behavior, Not Implementation`.

Basically mock your http requests such that it handles your edge cases (even if they are handled just in helper methods). You want test how a client uses your servlet directly. Nobody is calling your helper methods directly.

How to test HTTP servlet stuff (it even includes an example on how to test Writers. Maybe you could do something similar with Readers since you are using it in your implementation).
https://stackoverflow.com/questions/5434419/how-to-test-my-servlet-using-junit